### PR TITLE
Ming/book request for #27

### LINKED ISSUE #27: Developing Requesting Book Logic
## Description

This is going to be the underlying "backend" logic of what happens when a user requests a book. 
This ticket is going to involve:
1. Notifying users when their book has been requested
2. Allowing borrowers to request books
3. Allowing owners to accept request on books
4. Data consistency of the book/book requests

--- 
I won't write more because this is my own ticket anyway

### PR DESCRIPTION
**Pull request by mingyaang**
## Deliverable 

- User can request books
- User can accept book requests
- User can decline book requests
- User can get requested books
- User can get book requests

Please review this one more carefully since i fixed a lot of merge conflicts due to master being pushed into dev

closes #27

### CODE REVIEW

File: app/src/main/java/com/team41/boromi/MainActivity.java [hunk 1]
```diff
@@ -14,15 +15,18 @@
...
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     setContentView(R.layout.activity_main);
     ((BoromiApp) getApplicationContext()).appComponent.getAuthenticationComponent().inject(this);
 
+    // TODO REMOVE THIS SO IT DOESN'T REDIRECT TO TEST ACTIVITY
+    startActivity(new Intent(this, TestActivity.class));
+
```
Comment (akchen1) [inline app/src/main/java/com/team41/boromi/MainActivity.java]: I think should remove if merge to development

File: app/src/main/java/com/team41/boromi/adapters/PagerAdapter.java [hunk 2]
```diff
@@ -3,20 +3,23 @@
...
 import androidx.viewpager2.adapter.FragmentStateAdapter;
+
 import com.team41.boromi.FragmentFactory;
+
 import java.util.ArrayList;
 
 public class PagerAdapter extends FragmentStateAdapter {
 
-  private ArrayList<Pair<Class<? extends Fragment>, Bundle>> fragmentClasses;
+  private final ArrayList<Pair<Class<? extends Fragment>, Bundle>> fragmentClasses;
```
Comment (akchen1) [inline app/src/main/java/com/team41/boromi/adapters/PagerAdapter.java:19]: Why did you change to final?

File: app/src/main/java/com/team41/boromi/MainActivity.java (hunk 1 above)
Comment (mingyaang) [reply app/src/main/java/com/team41/boromi/MainActivity.java]: smd lol jk kinda true

File: app/src/main/java/com/team41/boromi/adapters/PagerAdapter.java (hunk 2 above)
Comment (mingyaang) [reply app/src/main/java/com/team41/boromi/adapters/PagerAdapter.java:19]: ide reeformatting :'(

**Review by akchen1 — APPROVED** [review]
lgtm
